File: cpp/2432. 处理用时最长的那个任务的员工.cpp
class Solution {
public:
    int hardestWorker(int n, vector<vector<int>>& logs) {
        int maxTime=0;
        int result=0;
        int t=0;
        for(int i=0;i<logs.size();i++){
            if(i==0) t=logs[i][1];
            else t=logs[i][1]-logs[i-1][1];
            if (t>maxTime){
                maxTime=t;
                result=logs[i][0];
            }else if (t==maxTime){
                result=min(result,logs[i][0]);
            }
        }
        return result;
    }
};

File: 刷题/golang/2432. 处理用时最长的那个任务的员工.go
func hardestWorker(n int, logs [][]int) int {
    maxTime,t := 0,0
    ans := 0
    for i, log :=range logs {
        if i==0 {
            t=log[1]
        }else {
            t=logs[i][1]-logs[i-1][1]
        }
        if t>maxTime {
            maxTime=t
            ans=log[0]
        }else if t==maxTime {
            ans=min(ans,log[0])
        }
    }
    return ans
}
func min(x, y int) int{
    if x>y {
        return y
    }
    return x
}